### DISCUSSION TITLE
Webservice mDNS discovery

### DISCUSSION THREAD

**Discussion by marciogranzotto** [NightscoutFoundation/xDrip#3154] [Ideas]
Could we add a mDNS service that registers the phone's ip address in the network?

I'm working on a ESP32 device with a display (M5 Paper) that will talk directly with the phone's IP to get glucose values and display them. The intent is to have it on my car's dashboard and connect it to the phone's wi-fi hotspot.
But I want to make it in a way that the ESP finds my phone in the network instead of hardcoding a IP address.

**Comment by jamorham** [comment]
If you can tell me a technical mechanism that wouldn't involve using up a lot of phone battery then I'd consider it. Can't you make the ESP32 use its default gateway address which will be the phone via hotspot?
